### PR TITLE
Drop Python 2.6 and 3.3 support

### LINKED ISSUE #5312: Remove support for Python 3.3
After #5283 ships in 0.21.0, we can remove support for Python 3.3. To do this, we should:

1. Appropriately update `setup.py` so our packages aren't listed as supporting Python 3.3.
2. Remove all Python 3.3 tests.
3. Remove Python 3.3 deprecation warnngs

### LINKED ISSUE #5489: Remove support for Python 2.6
Python 2.6 version of #5312.

### PR DESCRIPTION
**Pull request by bmw**
Built on #5508. Fixes #5489 and fixes #5312. This PR should be merged rather than squashed to preserve authorship.

@erikrose, do you have time to review this? Despite the size, it's fairly straight forward and you'll have the most context about setup.py/pipstrap changes.

### CODE REVIEW

**Comment by bmw** [comment]
To whoever reviews this PR, feel free to just review my commit. I'm happy with the changes in the first commit (with the exception of a few changes I reverted in mine).

File: letsencrypt-auto-source/letsencrypt-auto.template [hunk 1]
```diff
@@ -495,10 +504,18 @@ if [ "$1" = "--le-auto-phase2" ]; then
     say "Creating virtual environment..."
     DeterminePythonVersion
     rm -rf "$VENV_PATH"
-    if [ "$VERBOSE" = 1 ]; then
-      "$LE_PYTHON" -m venv "$VENV_PATH"
+    if [ "$PYVER" -le 27 ]; then
```
Comment (erikrose) [inline letsencrypt-auto-source/letsencrypt-auto.template]: Did you hang onto this branch on purpose?

Comment (bmw) [reply letsencrypt-auto-source/letsencrypt-auto.template]: Yeah. For most systems we're still using Python 2.7 and `python -m venv` is [new in Python 3.3+](https://docs.python.org/3/library/venv.html).

Comment (erikrose) [reply letsencrypt-auto-source/letsencrypt-auto.template]: Ah, I had a brain fart. <=2.7, not <2.7. r+!

**Pull request merged**